### PR TITLE
Update base images and remove obsolete 5.5 (fix #121)

### LINKED ISSUE #121: Use Debian:stretch-slim for newer images
While I still think there should be alpine variant of this image (#53), independently of whatever oracle does, the base debain image should be updated to use `debian:stretch-slim`, at the very least for `mariadb:10.3`. As this seems to be the new standard _debian_ base image most new versions of official images use.

### PR DESCRIPTION
**Pull request by J0WI**
5.5 is obsolete, 5.5.60 was the last release, so I just removed it.
> As per the MariaDB Deprecation Policy, this will be the last release of MariaDB 5.5 for Debian 7 Wheezy [[1]](https://mariadb.com/kb/en/library/mariadb-5560-release-notes/)

10.0 is only supported by Jessie, so I just changed it to the slim variant.

All other have been upgraded to `stretch-slim`. Furthermore they use a new GPG key (which is signed by the old one). `gnupg` can't be removed without running into huge issues with apt, so I just keep it installed.

`gpgkill` is only available on `stretch`. I didn't add it here to avoid additional overhead.

### CODE REVIEW

**Comment by tianon** [comment]
According to https://mariadb.com/kb/en/mariadb/what-is-mariadb-55/, the 5.5 release series is supported until April 2020, so I think we need to instead figure out a new base to switch this image to. :disappointed:

(It appears that Ubuntu Trusty is supported for the 5.5 series, which would line up well with the discussions elsewhere about switching to Xenial for better multiarchitecture support, but I'm not sure what that would entail given our odd dependency on Percona's repos too.)

**Comment by J0WI** [comment]
I found the following supported versions. Some of them are not supported until April 2020.

- OpenSUSE (**Jan 2019**)
  - amd64 (?)
- Ubuntu 14.04/Trusty (**Apr 2019**)
  - amd64
  - **i386**
  - **ppc64el**
- CentOS 6 (Nov 2020)
  - amd64
  - **i386**
- CentOS 7 (Jun 2024)
  - **aarch64**
  - amd64

Which do you prefer? :confused:

**Comment by yosifkit** [comment]
I'd lean toward Ubuntu Trusty for 5.5 and then 18.04 for the rest if we can. This will hopefully be the least breaking for users and get us more supported architectures from upstream MariaDB.  Perhaps add tags like `10.1-jessie` to the current images just to make the transition clear (like [python](https://hub.docker.com/_/python/) in https://github.com/docker-library/python/pull/215).

We still have to figure out what to do about `percona-xtrabackup`.  I think we might be best just installing it on the architectures it is possible for the older versions of MariaDB (and leaving unsupported arches like ppc64le without xtrabackup :cry:). Then we switch to `mariadb-backup` where it is supported. It looks like received GA status in 10.1.26+ and 10.2.10+ (https://mariadb.com/kb/en/library/mariadb-backup-overview/).

So basically a combination of this PR, #158 and #155.  I should have a chance to work on this in the coming weeks (feel free to update a PR or point to a branch you are working on if you want early review or collaboration).
